sklearn/linear_model/_bayes.py

```diff
@@ -553,17 +553,13 @@ def fit(self, X, y):
         self.scores_ = list()
         coef_old_ = None
 
-        # Compute sigma and mu (using Woodbury matrix identity)
         def update_sigma(X, alpha_, lambda_, keep_lambda, n_samples):
-            sigma_ = pinvh(np.eye(n_samples) / alpha_ +
-                           np.dot(X[:, keep_lambda] *
-                           np.reshape(1. / lambda_[keep_lambda], [1, -1]),
-                           X[:, keep_lambda].T))
-            sigma_ = np.dot(sigma_, X[:, keep_lambda] *
-                            np.reshape(1. / lambda_[keep_lambda], [1, -1]))
-            sigma_ = - np.dot(np.reshape(1. / lambda_[keep_lambda], [-1, 1]) *
-                              X[:, keep_lambda].T, sigma_)
-            sigma_.flat[::(sigma_.shape[1] + 1)] += 1. / lambda_[keep_lambda]
+            # See slides as referenced in the docstring note
+            X_keep = X[:, keep_lambda]
+            gram = np.dot(X_keep.T, X_keep)
+            eye = np.eye(gram.shape[0])
+            sigma_inv = lambda_[keep_lambda] * eye + alpha_ * gram
+            sigma_ = pinvh(sigma_inv)
```

Comment:
So that's basically the main fix. Instead of relying on the [woodbury formula](https://en.wikipedia.org/wiki/Woodbury_matrix_identity), we just directly invert the matrix (called S_N in  [the ref](http://www.utstat.toronto.edu/~rsalakhu/sta4273/notes/Lecture2.pdf#page=15)). This seems to be much more stable and also much faster because here we invert a matrix that is `n_features x features` whereas in master the inversion is on a `n_samples x n_samples`.

Strangely enough, this somewhat reverts a commit from 10 years ago https://github.com/scikit-learn/scikit-learn/commit/b2d0a77a11e0d6dd59c4f65a53d79be39678832c, whose goal was to make things faster but I'm not sure it did. @vmichel if you're still around your input would be greatly appreciated!

Reply:
could it be that numpy operations are now much faster?

Reply:
I doubt it, the bottleneck is the matrix inversion here